RHIDP-7560 - Install and configure plugins from Extensions page (#1281)
redhat-developer/red-hat-developers-documentation-rhdh: modules/dynamic-plugins/proc-extensions-configuring-rbac-role.adoc

[id="proc-extensions-configuring-rbac-role_{context}"]
= Configuring RBAC to manage Extensions

You can add Extensions permissions by creating or updating and existing RBAC role. For more information about using RBAC to manage role-based controls, see link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}/html-single/authorization_in_red_hat_developer_hub/managing-authorizations-by-using-the-web-ui[Managing role-based access controls (RBAC) using the {product} Web UI].

== Creating a role in the {product-short} UI to manage Extensions

.Prerequisites
* You have enabled RBAC, have a policy administrator role in {product-short}, and have added plugins with permission.

.Procedure

> . Go to Administration at the bottom of the sidebar in the {product-short}.

Reviewer: ```suggestion
. Go to *Administration* at the bottom of the sidebar in the {product-short}.
```

+

> The *RBAC* tab appears, displaying all the created roles in the Developer Hub.

Reviewer: ```suggestion
The *RBAC* tab appears, displaying all the created roles in the {product-short}.
```

. Click *Create* to create a role.
. Enter the user name and description (optional) of role in the given fields and click *Next*.
. In *Add users and groups*, select the user name, and click *Next*.
. In *Add permission policies*, select *Extensions* from the plugins dropdown.
. Expand *Extensions*, select both the *Create* and *Read* permissions for the Extensions plugin and click *Next*.
. Click *Create* to create the role.
+
image::rhdh/extensions-rbac-role-create.png[]

.Verification

> After you refresh the {product-very-short} application, when you select a plugin, the *Actions* dropdown will be active.

Reviewer 2: ```suggestion
After you refresh the {product-very-short} application, when you select a plugin, the *Actions* drop-down is active.
```

> When you click on the *Actions* drop-down, you can edit the the plugin configuration, and enable or disable the plugin.

Reviewer: ```suggestion
When you click the *Actions* drop-down, you can edit the the plugin configuration, and enable or disable the plugin.
```

> When you click on the *Active* dropdown, you can edit the the plugin configuration, and enable or disable the plugin.

Reviewer 2: ```suggestion
When you click on the *Active* drop-down, you can edit the the plugin configuration, and enable or disable the plugin.
```
According to[ IBM style guide](https://www.ibm.com/docs/en/ibm-style?topic=word-usage) in relation to GUI component, we can use "Drop-down".

// Not documented
// [source,yaml]
// ----
// p, role:default/team_a, extensions-plugin, read, allow
// p, role:default/team_a, extensions-plugin, create, allow
// g, user:default/<login-id/user-name>, role:default/team_a

// permission:
//       enabled: true
//       rbac:
//         admin:
//           users:

> //           - name: user:default/gerry-forde

Reviewer 3: Do we want to document how to set up these extensions permissions using csv file? I would say yes. We don't have to add the instructions to this page but would add them here: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/release-1.7/authorization/#ref-rbac-permission-policies_title-authorization

If we will be adding them to this page, please remove your name as a user and fix intendation.

Author: Created [ RHIDP-8584](https://issues.redhat.com/browse/RHIDP-8584) to provide this additional this documentation after the 1.7 release
